pygmt/tests/test_grdimage.py

```diff
@@ -51,3 +73,18 @@ def test_grdimage_fails():
     fig = Figure()
     with pytest.raises(GMTInvalidInput):
         fig.grdimage(np.arange(20).reshape((4, 5)))
+
+
+@pytest.mark.mpl_image_compare
+def test_grdimage_over_dateline(xrgrid):
+    """
+    Ensure no gaps are plotted over the 180 degree international dateline.
+    Specifically checking that `xrgrid.gmt.gtype = 1` sets `GMT_GRID_IS_GEO`,
+    and that `xrgrid.gmt.registration = 0` sets `GMT_GRID_NODE_REG`. Note that
+    there would be a gap over the dateline if a pixel registered grid is used.
+    """
+    fig = Figure()
+    assert xrgrid.gmt.registration == 0  # gridline registration
```

Comment:
This dateline test seems to work sometimes, but crashes at other times, reminds me of the black image we got a while ago. Was there a regression in GMT?

| ![baseline-test_grdimage_over_dateline](https://user-images.githubusercontent.com/23487320/87237486-fa0cfa80-c44a-11ea-99cf-eb2dc31c18de.png) | ![test_grdimage_over_dateline-failed-diff](https://user-images.githubusercontent.com/23487320/87237485-f8dbcd80-c44a-11ea-8343-18d0b8fa57f7.png) | ![test_grdimage_over_dateline](https://user-images.githubusercontent.com/23487320/87237487-faa59100-c44a-11ea-9085-1054c67da186.png)
|:---:|:---:|:---:|
| Expected | Diff | Generated |

Reply:
@weiji14 Please see if #517 works for you. 

You can cherry-pick commit b64f3713017749a80d704ef9683dd140b95b6ea7 to your local branch and see it can fix the random crashes.

Reply:
Nope, still get the all black image :/

Reply:
I ran the tests 10 times, and always get the expected figure, but I'm using macOS.

Reply:
Can you add `V='d'` to the grdimage function, and post the output so that I can compare if there are any differences.

Reply:
Yes, the macOS tests all pass, but if you look at the Ubuntu ones, only Python 3.8 passes (3.6 and 3.7 fails). Not sure about Windows.

Reply:
I get like 510 lines of `grdimage (gmtsupport_free_range): tried to free unallocated memory`. This is without cherry-picking #517 by the way, on Python 3.7:

<details>

```
grdimage [DEBUG]: Map distance calculation will be using great circle approximation with authalic auxiliary latitudes and authalic (R_2) radius = 6371007.1809 m, in meter.
grdimage [DEBUG]: Map distance calculation will be using great circle approximation with authalic auxiliary latitudes and authalic (R_2) radius = 6371007.1809 m, in meter.
grdimage [DEBUG]: Projected grid is non-orthogonal, nonlinear, or dpi was changed
grdimage [INFORMATION]: Read header from file @GMTAPI@-S-I-G-M-G-N-000000
grdimage [DEBUG]: gmtapi_begin_io: Input resource access is now enabled [container]
grdimage [DEBUG]: gmtapi_import_grid: Passed ID = 0 and mode = 1
grdimage [DEBUG]: Grid/Image dimensions imply w/e/s/n = 0/359/-89/90, inc = 1/1, gridline registration, n_layers = 1
grdimage [DEBUG]: Geographic input grid, longitudes span less than 360
grdimage [DEBUG]: Chosen boundary condition for all edges: geographic
grdimage [DEBUG]: Geographic input grid, longitude cells span exactly 360
grdimage [DEBUG]: Object ID 1 : Registered Grid Memory Reference 5593eda267a0 as an Input resource with geometry Surface [n_objects = 2]
grdimage [DEBUG]: Successfully created a new Grid container
grdimage [DEBUG]: GMT_End_IO: Input resource access is now disabled
grdimage [DEBUG]: Projected values in meters: -9.00996e+06 9.00996e+06 -9.00996e+06 9.00996e+06
grdimage [INFORMATION]: Map scale is 7094.46 km per cm or 1:7.09446e+08.
grdimage [DEBUG]: Running in PS mode modern
grdimage [DEBUG]: Use PS filename /home/user/.gmt/sessions/gmt_session.3726/gmt_6.ps-
grdimage [DEBUG]: Create hidden PS file /home/user/.gmt/sessions/gmt_session.3726/gmt_6.ps-
grdimage [DEBUG]: Got session name as pygmt-session and default graphics formats as pdf
grdimage [INFORMATION]: Allocate and read data from file @GMTAPI@-S-I-G-M-G-N-000000
grdimage [DEBUG]: gmtapi_begin_io: Input resource access is now enabled [container]
grdimage [DEBUG]: gmtapi_import_grid: Passed ID = 0 and mode = 2
grdimage [INFORMATION]: Importing grid data from user matrix memory location
grdimage [DEBUG]: Grid is considered to have a 360-degree longitude range.
grdimage [DEBUG]: Chosen boundary condition for left and right edges: geographic
grdimage [DEBUG]: Chosen boundary condition for bottom edge: natural
grdimage [DEBUG]: Chosen boundary condition for top    edge: geographic
grdimage [INFORMATION]: 359 (of 361) inconsistent grid values at North pole.
grdimage [INFORMATION]: Set boundary condition for top    edge: geographic
grdimage [INFORMATION]: Set boundary condition for bottom edge: natural
grdimage [DEBUG]: GMT_End_IO: Input resource access is now disabled
grdimage [INFORMATION]: project grid files
grdimage [DEBUG]: Object ID 2 : Registered Grid Memory Reference 5593ed9c3af0 as an Input resource with geometry Surface [n_objects = 3]
grdimage [DEBUG]: Successfully duplicated a Grid
grdimage [DEBUG]: gmt_project_init: IN: Inc [0/0] n_columns/n_rows [361/180] dpi = 0 offset = 0
grdimage [DEBUG]: gmt_project_init: OUT: Inc [0/0] n_columns/n_rows [361/180] dpi = 0 offset = 0
grdimage [INFORMATION]: Grid projection from size 361x180 to 361x180
grdimage [DEBUG]: Successfully added data array to previously registered Grid container
grdimage [DEBUG]: gmt_grd_project: In [0/360/-89/90] and out [0/1/0/1]
grdimage [INFORMATION]: gmt_grd_project: Output grid extrema [-27.5593/114.589] exceed extrema of input grid [-1/1] due to resampling
grdimage [INFORMATION]: gmt_grd_project: Output grid clipped to input grid extrema
grdimage [DEBUG]: GMT_Destroy_Data: freed memory for a Grid for object 1
grdimage [DEBUG]: gmtlib_unregister_io: Unregistering object no 1 [n_objects = 2]
grdimage [DEBUG]: gmtlib_unregister_io: Object no 1 has non-NULL resource pointer
grdimage [DEBUG]: gmtlib_get_graphics_item: Fig: 6 Subplot: 2 Panel: () Inset: 0
grdimage [DEBUG]: No current cpt file found
grdimage [DEBUG]: gmtlib_get_graphics_item: Fig: 6 Subplot: 2 Panel: () Inset: 0
grdimage [DEBUG]: No current cpt file found
grdimage [DEBUG]: GMT: 0. Will try to find subdir=cpt stem = turbo suffix=.cpt
grdimage [DEBUG]: GMT: 1. gmt_getsharepath trying current dir
grdimage [DEBUG]: GMT: 2. gmt_getsharepath trying USERDIR /home/user/.gmt
grdimage [DEBUG]: GMT: 3. gmt_getsharepath trying USERDIR subdir /home/user/.gmt/cpt
grdimage [DEBUG]: GMT: 4. gmt_getsharepath trying SHAREDIR subdir /home/user/miniconda3/envs/pygmt/share/gmt/cpt
grdimage [DEBUG]: gmt_get_filename: In: /home/user/miniconda3/envs/pygmt/share/gmt/cpt/turbo.cpt Out: /home/user/miniconda3/envs/pygmt/share/gmt/cpt/turbo.cpt
grdimage [DEBUG]: Given full path to file /home/user/miniconda3/envs/pygmt/share/gmt/cpt/turbo.cpt
grdimage [DEBUG]: gmt_get_filename: In: /home/user/miniconda3/envs/pygmt/share/gmt/cpt/turbo.cpt Out: /home/user/miniconda3/envs/pygmt/share/gmt/cpt/turbo.cpt
grdimage [DEBUG]: gmt_get_filename: In: /home/user/miniconda3/envs/pygmt/share/gmt/cpt/turbo.cpt Out: /home/user/miniconda3/envs/pygmt/share/gmt/cpt/turbo.cpt
grdimage [DEBUG]: Found readable file /home/user/miniconda3/envs/pygmt/share/gmt/cpt/turbo.cpt
grdimage [DEBUG]: Object ID 3 : Registered CPT File /home/user/miniconda3/envs/pygmt/share/gmt/cpt/turbo.cpt as an Input resource with geometry Non-Geographical [n_objects = 3]
grdimage [DEBUG]: gmtapi_begin_io: Input resource access is now enabled [container]
grdimage [DEBUG]: gmtapi_import_palette: Passed ID = 3 and mode = 8
grdimage [INFORMATION]: Reading CPT from File /home/user/miniconda3/envs/pygmt/share/gmt/cpt/turbo.cpt
grdimage [DEBUG]: Reading CPT from /home/user/miniconda3/envs/pygmt/share/gmt/cpt/turbo.cpt
grdimage [DEBUG]: GMT_End_IO: Input resource access is now disabled
grdimage [INFORMATION]: Auto-stretching CPT file turbo to fit data range -1 to 1
grdimage [DEBUG]: gmtlib_get_graphics_item: Fig: 6 Subplot: 2 Panel: () Inset: 0
grdimage [DEBUG]: Object ID 4 : Registered CPT File /home/user/.gmt/sessions/gmt_session.3726/gmt.6.cpt as an Output resource with geometry Non-Geographical [n_objects = 4]
grdimage [DEBUG]: gmtapi_begin_io: Output resource access is now enabled [container]
grdimage [DEBUG]: gmtapi_export_palette: Passed ID = 4 and mode = 0
grdimage [INFORMATION]: Write CPT to File /home/user/.gmt/sessions/gmt_session.3726/gmt.6.cpt
grdimage [DEBUG]: Writing CPT to /home/user/.gmt/sessions/gmt_session.3726/gmt.6.cpt
grdimage [DEBUG]: GMT_End_IO: Output resource access is now disabled
grdimage [INFORMATION]: Save current CPT file to /home/user/.gmt/sessions/gmt_session.3726/gmt.6.cpt !
grdimage [INFORMATION]: Evaluate image pixel colors
grdimage [INFORMATION]: Basic z(x,y) with optional illumination and no PostScript colormasking.
grdimage [INFORMATION]: Creating 24-bit color image
grdimage [DEBUG]: Current size of half-baked PS file /home/user/.gmt/sessions/gmt_session.3726/gmt_6.ps- = 55422.
grdimage (GMT_grdimage): tried to free unallocated memory
grdimage [DEBUG]: GMT_Destroy_Data: freed memory for a Grid for object 2
grdimage [DEBUG]: gmtlib_unregister_io: Unregistering object no 2 [n_objects = 3]
grdimage [DEBUG]: gmtlib_unregister_io: Object no 2 has non-NULL resource pointer
grdimage [DEBUG]: gmtlib_garbage_collection: Destroying object: C=0 A=0 ID=3 W=Input F=CPT M=File S=Used P=5593edb86930 N=/home/user/miniconda3/envs/pygmt/share/gmt/cpt/turbo.cpt
grdimage (gmtsupport_free_range): tried to free unallocated memory
grdimage (gmtsupport_free_range): tried to free unallocated memory
...
grdimage (gmtsupport_free_range): tried to free unallocated memory
grdimage (gmtsupport_free_range): tried to free unallocated memory
grdimage [DEBUG]: GMTAPI_Garbage_Collection freed 1 memory objects
grdimage [DEBUG]: gmtlib_unregister_io: Unregistering object no 3 [n_objects = 2]
grdimage [DEBUG]: gmtlib_unregister_io: Unregistering object no 4 [n_objects = 1]
grdimage (gmtlib_free_tmp_arrays): tried to free unallocated memory
```

</details>

Reply:
I don't see any difference between macOS and Linux.

Reply:
Hold on, that error message was from one of the successful runs. It seems to work fine in ipython but fails when I run it using `make test`. Give me a couple of minutes.

Reply:
Ok, got the failing debug info (not sure why it's not printed in the CI). The main difference seems to be an `inf` value:
```
# good
grdimage [INFORMATION]: gmt_grd_project: Output grid extrema [-27.5593/114.589] exceed extrema of input grid [-1/1] due to resampling
```

```
# bad
grdimage [INFORMATION]: gmt_grd_project: Output grid extrema [-inf/114.589] exceed extrema of input grid [-inf/1] due to resampling
```

<details>

```
grdimage [DEBUG]: Map distance calculation will be using great circle approximation with authalic auxiliary latitudes and authalic (R_2) radius = 6371007.1809 m, in meter.
grdimage [DEBUG]: Map distance calculation will be using great circle approximation with authalic auxiliary latitudes and authalic (R_2) radius = 6371007.1809 m, in meter.
grdimage [DEBUG]: Projected grid is non-orthogonal, nonlinear, or dpi was changed
grdimage [INFORMATION]: Read header from file @GMTAPI@-S-I-G-M-G-N-000000
grdimage [DEBUG]: gmtapi_begin_io: Input resource access is now enabled [container]
grdimage [DEBUG]: gmtapi_import_grid: Passed ID = 0 and mode = 1
grdimage [DEBUG]: Grid/Image dimensions imply w/e/s/n = 0/359/-89/90, inc = 1/1, gridline registration, n_layers = 1
grdimage [DEBUG]: Geographic input grid, longitudes span less than 360
grdimage [DEBUG]: Chosen boundary condition for all edges: geographic
grdimage [DEBUG]: Geographic input grid, longitude cells span exactly 360
grdimage [DEBUG]: Object ID 1 : Registered Grid Memory Reference 562abb2b9ce0 as an Input resource with geometry Surface [n_objects = 2]
grdimage [DEBUG]: Successfully created a new Grid container
grdimage [DEBUG]: GMT_End_IO: Input resource access is now disabled
grdimage [DEBUG]: Projected values in meters: -9.00996e+06 9.00996e+06 -9.00996e+06 9.00996e+06
grdimage [INFORMATION]: Map scale is 7094.46 km per cm or 1:7.09446e+08.
grdimage [DEBUG]: Running in PS mode modern
grdimage [DEBUG]: Use PS filename /home/user/.gmt/sessions/gmt_session.6871/gmt_68.ps-
grdimage [DEBUG]: Create hidden PS file /home/user/.gmt/sessions/gmt_session.6871/gmt_68.ps-
grdimage [DEBUG]: Got session name as pygmt-session and default graphics formats as pdf
grdimage [INFORMATION]: Allocate and read data from file @GMTAPI@-S-I-G-M-G-N-000000
grdimage [DEBUG]: gmtapi_begin_io: Input resource access is now enabled [container]
grdimage [DEBUG]: gmtapi_import_grid: Passed ID = 0 and mode = 2
grdimage [INFORMATION]: Importing grid data from user matrix memory location
grdimage [DEBUG]: Grid is considered to have a 360-degree longitude range.
grdimage [DEBUG]: Chosen boundary condition for left and right edges: geographic
grdimage [DEBUG]: Chosen boundary condition for bottom edge: natural
grdimage [DEBUG]: Chosen boundary condition for top    edge: geographic
grdimage [INFORMATION]: 359 (of 361) inconsistent grid values at North pole.
grdimage [INFORMATION]: Set boundary condition for top    edge: geographic
grdimage [INFORMATION]: Set boundary condition for bottom edge: natural
grdimage [DEBUG]: GMT_End_IO: Input resource access is now disabled
grdimage [INFORMATION]: project grid files
grdimage [DEBUG]: Object ID 2 : Registered Grid Memory Reference 562abb070e10 as an Input resource with geometry Surface [n_objects = 3]
grdimage [DEBUG]: Successfully duplicated a Grid
grdimage [DEBUG]: gmt_project_init: IN: Inc [0/0] n_columns/n_rows [361/180] dpi = 0 offset = 0
grdimage [DEBUG]: gmt_project_init: OUT: Inc [0/0] n_columns/n_rows [361/180] dpi = 0 offset = 0
grdimage [INFORMATION]: Grid projection from size 361x180 to 361x180
grdimage [DEBUG]: Successfully added data array to previously registered Grid container
grdimage [DEBUG]: gmt_grd_project: In [0/360/-89/90] and out [0/1/0/1]
grdimage [INFORMATION]: gmt_grd_project: Output grid extrema [-inf/114.589] exceed extrema of input grid [-inf/1] due to resampling
grdimage [INFORMATION]: gmt_grd_project: Output grid clipped to input grid extrema
grdimage [DEBUG]: GMT_Destroy_Data: freed memory for a Grid for object 1
grdimage [DEBUG]: gmtlib_unregister_io: Unregistering object no 1 [n_objects = 2]
grdimage [DEBUG]: gmtlib_unregister_io: Object no 1 has non-NULL resource pointer
grdimage [DEBUG]: gmtlib_get_graphics_item: Fig: 68 Subplot: 2 Panel: () Inset: 0
grdimage [DEBUG]: No current cpt file found
grdimage [DEBUG]: gmtlib_get_graphics_item: Fig: 68 Subplot: 2 Panel: () Inset: 0
grdimage [DEBUG]: No current cpt file found
grdimage [DEBUG]: GMT: 0. Will try to find subdir=cpt stem = turbo suffix=.cpt
grdimage [DEBUG]: GMT: 1. gmt_getsharepath trying current dir
grdimage [DEBUG]: GMT: 2. gmt_getsharepath trying USERDIR /home/user/.gmt
grdimage [DEBUG]: GMT: 3. gmt_getsharepath trying USERDIR subdir /home/user/.gmt/cpt
grdimage [DEBUG]: GMT: 4. gmt_getsharepath trying SHAREDIR subdir /home/user/miniconda3/envs/pygmt/share/gmt/cpt
grdimage [DEBUG]: gmt_get_filename: In: /home/user/miniconda3/envs/pygmt/share/gmt/cpt/turbo.cpt Out: /home/user/miniconda3/envs/pygmt/share/gmt/cpt/turbo.cpt
grdimage [DEBUG]: Given full path to file /home/user/miniconda3/envs/pygmt/share/gmt/cpt/turbo.cpt
grdimage [DEBUG]: gmt_get_filename: In: /home/user/miniconda3/envs/pygmt/share/gmt/cpt/turbo.cpt Out: /home/user/miniconda3/envs/pygmt/share/gmt/cpt/turbo.cpt
grdimage [DEBUG]: gmt_get_filename: In: /home/user/miniconda3/envs/pygmt/share/gmt/cpt/turbo.cpt Out: /home/user/miniconda3/envs/pygmt/share/gmt/cpt/turbo.cpt
grdimage [DEBUG]: Found readable file /home/user/miniconda3/envs/pygmt/share/gmt/cpt/turbo.cpt
grdimage [DEBUG]: Object ID 3 : Registered CPT File /home/user/miniconda3/envs/pygmt/share/gmt/cpt/turbo.cpt as an Input resource with geometry Non-Geographical [n_objects = 3]
grdimage [DEBUG]: gmtapi_begin_io: Input resource access is now enabled [container]
grdimage [DEBUG]: gmtapi_import_palette: Passed ID = 3 and mode = 8
grdimage [INFORMATION]: Reading CPT from File /home/user/miniconda3/envs/pygmt/share/gmt/cpt/turbo.cpt
grdimage [DEBUG]: Reading CPT from /home/user/miniconda3/envs/pygmt/share/gmt/cpt/turbo.cpt
grdimage [DEBUG]: GMT_End_IO: Input resource access is now disabled
grdimage [INFORMATION]: Auto-stretching CPT file turbo to fit data range -inf to 1
grdimage [DEBUG]: gmtlib_get_graphics_item: Fig: 68 Subplot: 2 Panel: () Inset: 0
grdimage [DEBUG]: Object ID 4 : Registered CPT File /home/user/.gmt/sessions/gmt_session.6871/gmt.68.cpt as an Output resource with geometry Non-Geographical [n_objects = 4]
grdimage [DEBUG]: gmtapi_begin_io: Output resource access is now enabled [container]
grdimage [DEBUG]: gmtapi_export_palette: Passed ID = 4 and mode = 0
grdimage [INFORMATION]: Write CPT to File /home/user/.gmt/sessions/gmt_session.6871/gmt.68.cpt
grdimage [DEBUG]: Writing CPT to /home/user/.gmt/sessions/gmt_session.6871/gmt.68.cpt
grdimage [DEBUG]: GMT_End_IO: Output resource access is now disabled
grdimage [INFORMATION]: Save current CPT file to /home/user/.gmt/sessions/gmt_session.6871/gmt.68.cpt !
grdimage [INFORMATION]: Evaluate image pixel colors
grdimage [INFORMATION]: Basic z(x,y) with optional illumination and no PostScript colormasking.
grdimage [INFORMATION]: Creating 24-bit color image
PSL: Colormap of 2 colors created
PSL: Image depth reduced to 1 bits
PSL: DEFLATE compressed 8280 to 668 bytes (91.9% savings at compression level 5)
grdimage [DEBUG]: Current size of half-baked PS file /home/user/.gmt/sessions/gmt_session.6871/gmt_68.ps- = 22240.
grdimage (GMT_grdimage): tried to free unallocated memory
grdimage [DEBUG]: GMT_Destroy_Data: freed memory for a Grid for object 2
grdimage [DEBUG]: gmtlib_unregister_io: Unregistering object no 2 [n_objects = 3]
grdimage [DEBUG]: gmtlib_unregister_io: Object no 2 has non-NULL resource pointer
grdimage [DEBUG]: gmtlib_garbage_collection: Destroying object: C=0 A=0 ID=3 W=Input F=CPT M=File S=Used P=562abf53bc60 N=/home/user/miniconda3/envs/pygmt/share/gmt/cpt/turbo.cpt
grdimage (gmtsupport_free_range): tried to free unallocated memory
grdimage (gmtsupport_free_range): tried to free unallocated memory
...
grdimage (gmtsupport_free_range): tried to free unallocated memory
grdimage (gmtsupport_free_range): tried to free unallocated memory
grdimage [DEBUG]: GMTAPI_Garbage_Collection freed 1 memory objects
grdimage [DEBUG]: gmtlib_unregister_io: Unregistering object no 3 [n_objects = 2]
grdimage [DEBUG]: gmtlib_unregister_io: Unregistering object no 4 [n_objects = 1]
grdimage (gmtlib_free_tmp_arrays): tried to free unallocated memory
```

</details>

Reply:
Yes, I saw the similar *inf* when debugging the `shading=True` failure in https://github.com/GenericMappingTools/pygmt/issues/364. That's why I thought `GMT_IN` should work.

Reply:
It could be another potential GMT bug, which is usually difficult to debugging from the PyGMT side.

Reply:
FYI, this is the output using `GMT_IS_DUPLICATE`, which generates the reddish-brown image above. Using `V="i"` for brevity, note that `-1.1949e+26` is printed instead of `-inf`:

```
grdimage [INFORMATION]: Read header from file @GMTAPI@-S-I-G-M-G-N-000000
grdimage [INFORMATION]: Map scale is 7094.46 km per cm or 1:7.09446e+08.
grdimage [INFORMATION]: Allocate and read data from file @GMTAPI@-S-I-G-M-G-N-000000
grdimage [INFORMATION]: Importing grid data from user matrix memory location
grdimage [INFORMATION]: 359 (of 361) inconsistent grid values at North pole.
grdimage [INFORMATION]: Set boundary condition for top    edge: geographic
grdimage [INFORMATION]: Set boundary condition for bottom edge: natural
grdimage [INFORMATION]: project grid files
grdimage [INFORMATION]: Grid projection from size 361x180 to 361x180
grdimage [INFORMATION]: gmt_grd_project: Output grid extrema [-1.1949e+26/114.589] exceed extrema of input grid [-1.1949e+26/1] due to resampling
grdimage [INFORMATION]: gmt_grd_project: Output grid clipped to input grid extrema
grdimage [INFORMATION]: Reading CPT from File /home/user/miniconda3/envs/pygmt/share/gmt/cpt/turbo.cpt
grdimage [INFORMATION]: Auto-stretching CPT file turbo to fit data range -1.1949e+26 to 1
grdimage [INFORMATION]: Write CPT to File /home/user/.gmt/sessions/gmt_session.12777/gmt.68.cpt
grdimage [INFORMATION]: Save current CPT file to /home/user/.gmt/sessions/gmt_session.12777/gmt.68.cpt !
grdimage [INFORMATION]: Evaluate image pixel colors
grdimage [INFORMATION]: Basic z(x,y) with optional illumination and no PostScript colormasking.
grdimage [INFORMATION]: Creating 24-bit color image
PSL: Colormap of 3 colors created
PSL: Image depth reduced to 2 bits
PSL: DEFLATE compressed 16380 to 779 bytes (95.2% savings at compression level 5)
```